app/helpers/deliverers_helper.rb

```diff
@@ -1,2 +1,42 @@
 module DeliverersHelper
+  def month_range
+    today = Time.zone.today
+    (today - 29)..today
+  end
+
+  # :reek:TooManyStatements
+  def shifts_on_date(queue, date)
```

Comment:
try using `group_by` to group shifts by date to simplify this operation
https://apidock.com/ruby/Enumerable/group_by

Reply:
Fixed, now using `group_by` with Date arithmetic